frontend/src/App.vue

```diff
@@ -54,16 +53,67 @@ export default {
       );
     },
   },
-  watch: {
-  },
+  watch: {},
   methods: {
     async onAction(contextualItem, index) {
       const itemFqid = contextualItem?.item?.fqid;
-      await this.rpc.invoke("performAction", [itemFqid, index, contextualItem.context]);
+      await this.rpc.invoke("performAction", [
+        itemFqid,
+        index,
+        contextualItem.context,
+      ]);
     },
+  getItem(itemFqid) {
+    for (const collection of this.collections) {
+      for (const item of collection.items) {
+        if (item.fqid === itemFqid) {
+          return item;
+        }
+        if (item.items) {
+          for (const subItem of item.items) {
+            if (subItem.fqid === itemFqid) {
+              return subItem;
+            }  
+            else if( subItem.items ){
```

Comment:
please format this `else if` with the same style as other `if` conditions in this method